enhancements/machine-api/converting-machine-api-to-cluster-api.md

```diff
@@ -0,0 +1,756 @@
+we can fail the synchronisation and add a condition to the Machine API resource to identify that the feature is not yet supported in Cluster API.
+
+When these features are identified, we must prioritise adding the feature in Cluster API as these will become blockers for migration.
+Once the feature is implemented in Cluster API, we can unblock the synchronisation for the feature and allow users to migrate.
+
+#### Multiple controllers reconcile the same infrastructure
+
+We aim to prevent multiple controllers (being Machine API and Cluster API) simultaneously reconciling infrastructure as described in the [filtering][filtering] above.
+
+However, were there to be multiple controllers reconciling the same Machine, there are three cases to consider. Create, Update and Deletion operations.
+
+In a Create operation, if both controllers attempted the create simultaneously, it could lead to an additional host being created and then orphaned within the infrastructure provider.
+Once a controller has created the host, it is expected to populate the provider ID and use this to look up the infrastructure in future reconciles.
+The synchronisation controller would overwrite the non-authoritative provider ID, effectively orphaning the host.
+
+In an Update operation, since the infrastructure is immutable, the updates are only gathering information.
+In this case, the synchronisation controller would overwrite the information, should it differ, in the non-authoritative API.
+This could lead to a hot loop/fight between the two controllers.
+Rate limiting should be used in the controllers to prevent rapid reconciles in this fashion.
+
+In a Delete operation, both controllers would attempt to delete the same infrastructure.
+It is expected that this would not cause issue in most cases as the infrastructure provider should be able to handle multiple requests to delete the same object.
+Where Machine lifecycle hooks are in place on Machines, there should be mirrored across both APIs and should prevent either taking action until they are removed.
+Machine lifecycle hooks are expected to continue to function as expected.
+
+From this, it is imperative that we ensure Create operations are not reconciled by multiple Machine controllers to prevent the potential of leaking resources.
+Testing will be focused on the creation flow.
+
+[filtering]: #filtering-reconciles-in-capi-controllers
+
+### Drawbacks
+
+#### Carrying filtering logic in the Cluster API controllers
+
+We aim to carry as little code in forks as possible, however, we are changing the behaviour of the controllers with the reconcile filtering logic described above.
+
+This will be a fairly major patch that we need to carry.
+We need to make sure that the patch will apply cleanly in the future and that we minimise the maintenance burden of this carry as much as possible.
+
+#### Mirroring resources may be confusing for end users
```

Comment:
I would personally say we should add a condition to the CAPI resource rather than an annotation. Other controllers _should_ leave the additional condition alone. Will need to be tested though

Reply:
We are now proposing to add a Paused condition upstream, do you think this would cover this concern? We intend to use pausing upstream as a way to stop CAPI from reconciling the resources, and with admission control, will prevent the unpausing of resources

Reply:
Yes, I think pausing on CAPI plus the admission control would help.